supplier_product/models.py

```diff
@@ -0,0 +1,59 @@
+from django.db import models
+from django.core.validators import MinLengthValidator, MinValueValidator
+from product.models import Product
+
+
+class User(models.Model):
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    """Supplier class inherit from User abstract class.
+    Supplier class lets the app user to do variety of actions
+    which evolves the Product and Delivery class.
+    Attributes:
+        business_name: a string which holds the Supplier business name.
+    """
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey('Supplier', on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField(validators=[MinValueValidator(0)])
+
+    class Meta:
+        unique_together = ('suppler_product_id', 'qr_code')
+
+    def increase_sup_product_quantity(self):
+        self.quantity += 1
+        self.save()
+
+    def decrease_sup_product_quantity(self):
```

Comment:
```suggestion
    def decrease_sup_product_quantity(self):
    if self.quantity == 1:
        delete the row
    else:
      remove 1 from the quantity
```
We want to match this logic in the increase and decrease in order to get correct results